renderer/components/Settings/SettingsFieldsWallet.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Bar, DataRow, Select } from 'components/UI'
+import { FieldLabel, NumberField } from './SettingsFieldHelpers'
+import messages from './messages'
+
+const addressItems = [{ key: 'p2wkh' }, { key: 'np2wkh' }]
+const addressMessageMapper = key => {
+  const filters = {
+    p2wkh: messages.address_option_p2wkh,
+    np2wkh: messages.address_option_np2wkh,
+  }
+  return filters[key]
+}
+
+const SettingsFieldsWallet = ({ currentConfig }) => {
+  return (
+    <>
+      <DataRow
+        left={<FieldLabel itemKey="address" />}
+        right={
+          <Select
+            field="address"
+            highlightOnValid={false}
+            initialSelectedItem={currentConfig.address}
+            items={addressItems}
+            messageMapper={addressMessageMapper}
+          />
+        }
+      />
+
+      <Bar variant="light" />
+
+      <DataRow
```

Comment:
Could be a good extraction, since it's basically `fn(targetConf)`

Reply:
I was referring to ones that use `NumberField`. Is see 3 similar ones. Could be `NumberDataRow`

Reply:
Ok yes that actually reads much better. Have updated.